### PR TITLE
fix: associate learn lessons with all ecosystems [IDE-203]

### PR DESCRIPTION
**Pull request by bastiandoetsch**
### Description

Previously, a lesson was only associated with one ecosystem. Now it is associated with all ecosystems.

### Checklist

- [x] Tests added and all succeed
- [x] Linted
- [ ] README.md updated, if user-facing
- [ ] License file updated, if new 3rd-party dependency is introduced

🚨After having merged, please update the CLI go.mod to pull in latest language server.

### Screenshots / GIFs

_Visuals that may help the reviewer. Please add screenshots for any UI change. GIFs are most welcome!_

### CODE REVIEW

File: infrastructure/learn/service.go [hunk 1]
```diff
@@ -204,13 +204,14 @@ func (s *serviceImpl) updateCaches(lessons []Lesson) {
...
+			lessonsForEcosystem, exists := s.lessonsByEcosystemCache.Get(ecosystem)
+			if !exists {
+				lessonsForEcosystem = []Lesson{}
+			}
+			lessonsForEcosystem = append(lessonsForEcosystem, lesson)
+			s.lessonsByEcosystemCache.Set(ecosystem, lessonsForEcosystem, expiration)
 		}
-		lessonsForEcosystem = append(lessonsForEcosystem, lesson)
-		s.lessonsByEcosystemCache.Set(lesson.Ecosystem, lessonsForEcosystem, expiration)
 	}
```
Comment (cat2608) [inline infrastructure/learn/service.go:215]: I'm not sure if it's related to this change or it's a random issue but I observed the following:
|main|fix/snyk-learn-for-all-ecosystems|
|-|-|
|main-with-link|missing-link|

**Review by cat2608 — APPROVED** [review]
I've tested it locally and for Code Quality, it seems that we have a regression?

File: infrastructure/learn/service.go (hunk 1 above)
Comment (bastiandoetsch) [reply infrastructure/learn/service.go:215]: Either a regression or it was wrong before. I'll have a look!

Comment (bastiandoetsch) [reply infrastructure/learn/service.go:215]: Reproduced it, it was a bug before - it was linking to the Path Traversal learn lesson, which it shouldn't have.

**Pull request merged**